GenericsHomework/Node.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace GenericsHomework
+{
+    public class Node<T>
+    {
+        public T? Data { get; set; }
+        private int Size { get; set; }
+        public Node<T> Next { get; set; }
+
+        public Node(T? data)
+        {
+
+            this.Data = data;
+            this.Next = this;
+        }
+        public override string ToString()
+        {
+            return Data?.ToString() ?? "null";
+        }
+        public void Append(T data)
+        {
+            if (data == null) return;
+            if(existing(data) == true)
+            {
+                throw new ArgumentException();
```

Comment:
```suggestion
                throw new ArgumentException("Cannot add duplicate item");
```
You should add some kind of message here to let the user know what's wrong, probably something better than what I wrote... But you get the idea. :)